cpp/src/arrow/flight/CMakeLists.txt

```diff
@@ -231,6 +231,17 @@ foreach(LIB_TARGET ${ARROW_FLIGHT_LIBRARIES})
   target_compile_definitions(${LIB_TARGET} PRIVATE ARROW_FLIGHT_EXPORTING)
 endforeach()
 
+# Handle Unity build header conflicts on Windows
```

Comment:
@kou what do you think of this? Should we use a different approach to fix the windows unity builds?

Reply:
Can I see the error log?

How about `#include <arrow/flight/sql/odbc/odbcabstraction/include/odbcabstraction/platform.h>` in all `arrow/flight/sql/odbc/**.cc` instead?

Reply:
Is not on odbc is on normal arrow flight see the comment above for the original error:
https://github.com/apache/arrow/pull/47627#issuecomment-3334160407

Reply:
Hmm... The error message doesn't include source file information...

In general, we should skip unity build for the target source files by `SKIP_UNITY_BUILD_INCLUSION TRUE` like https://github.com/apache/arrow/blob/13c2615f6772eda028c57f5f99bdcbf708126ff0/cpp/src/arrow/flight/CMakeLists.txt#L122-L124 .

But the error message doesn't provide hints for the target source files...

Reply:
Yes, I wasn't able to find out what the problematic source file was with all my tests. Non-unity builds works well, I tested different things and the only thing I found to work was to force include those headers. I would love to upgrade our vcpkg version now that this works for all wheels (mainly because of the outdated version and the fix to the macos wheels).